crates/storage/store_db/libmdbx.rs

```diff
@@ -349,9 +363,19 @@ impl StoreEngine for Store {
         let cursor = txn
             .cursor::<TransactionLocations>()
             .map_err(StoreError::LibmdbxError)?;
-        Ok(cursor
-            .walk_key(transaction_hash.into(), None)
-            .map_while(|res| res.ok().map(|t| t.to()))
+
+        let mut transaction_hashes = Vec::new();
+        for res in cursor.walk_key(transaction_hash.into(), None) {
+            match res {
+                Ok(res) => {
+                    transaction_hashes.push(res.to().map_err(StoreError::from)?);
+                }
+                Err(_) => break,
+            }
```

Comment:
If we don't care about the error there's no need to `match` here. Use `if let` instead.
```suggestion
            if let Ok(res) = res {
                transaction_hashes.push(res.to().map_err(StoreError::from)?);
            } else {
                break;
            }
```